frontend/www/js/omegaup/contest/edit.ts

```diff
@@ -241,6 +285,18 @@ OmegaUp.on('ready', () => {
             problem: types.ProblemsetProblem;
             isUpdate: boolean;
           }) => {
+            api.Problem.details({
+              problem_alias: problem.alias,
+            })
+              .then((problemDetails) => {
+                if (
+                  problemDetails.languages.includes('cat') &&
+                  !this.details.languages.includes('cat')
+                ) {
+                  this.addCatToLanguages();
+                }
+              })
+              .catch(ui.apiError);
```

Comment:
esto funciona?

```suggestion
            if (problem.languages.split(',').includes('cat') &&
                !payload.details.languages.includes('cat')) {
              api.Contest.update({
                contest_alias: payload.details.alias,
                languages: payload.details.languages.concat(['cat']),
              })
                .then(() => {
                  ui.warning(T.contestEditCatLanguageAddedWarning);
                })
                .catch(ui.apiError);
            }
```

en teoría el `problem` ya incluye los lenguajes que necesita, así que no debería haber necesidad de mandar llamar un API para re-obtenerlos. además, el API para actualizar sólo necesita un campo, entonces tampoco hay necesidad de crear una función nada más para llamar un API.

Reply:
oh, `problem` no incluye los lenguajes, porque creo que sólo se pasa el alias de los problemas para el `typeahead`, y cuando el usuario selecciona un problema para agregar, para mostrar las versiones del problema se llama al API `Problem.versions`, por eso había pensando en hacer lo mismo con los lenguajes. 
Entonces ahora mejor actualice los lenguajes del problema después de agregar el problema al concurso y _después_ de actualizar la lista de problemas, para así obtener los lenguajes del último problema añadido.